direct/tpm2/combined_context_test.go

```diff
@@ -0,0 +1,97 @@
+package tpm2
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-tpm/direct/structures/tpm"
+	"github.com/google/go-tpm/direct/structures/tpm2b"
+	"github.com/google/go-tpm/direct/structures/tpma"
+	"github.com/google/go-tpm/direct/structures/tpml"
+	"github.com/google/go-tpm/direct/structures/tpms"
+	"github.com/google/go-tpm/direct/structures/tpmt"
+	"github.com/google/go-tpm/direct/structures/tpmu"
+	"github.com/google/go-tpm/direct/transport/simulator"
+)
+
+func TestCombinedContext(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	PCR7, err := CreatePCRSelection([]int{7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+
+	createPrimary := CreatePrimary{
+		PrimaryHandle: tpm.RHOwner,
+
+		InPublic: tpm2b.Public{
+			PublicArea: tpmt.Public{
+				Type:    tpm.AlgRSA,
+				NameAlg: tpm.AlgSHA256,
+				ObjectAttributes: tpma.Object{
+					SignEncrypt:         true,
+					FixedTPM:            true,
+					FixedParent:         true,
+					SensitiveDataOrigin: true,
+					UserWithAuth:        true,
+				},
+				Parameters: tpmu.PublicParms{
+					RSADetail: &tpms.RSAParms{
+						Scheme: tpmt.RSAScheme{
+							Scheme: tpm.AlgRSASSA,
+							Details: tpmu.AsymScheme{
+								RSASSA: &tpms.SigSchemeRSASSA{
+									HashAlg: tpm.AlgSHA256,
+								},
+							},
+						},
+						KeyBits: 2048,
+					},
+				},
+			},
+		},
+		CreationPCR: tpml.PCRSelection{
+			PCRSelections: []tpms.PCRSelection{
+				{
+					Hash:      tpm.AlgSHA1,
+					PCRSelect: PCR7,
+				},
+			},
+		},
+	}
+
+	rspCP, err := createPrimary.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("could not create key: %v", err)
+	}
+
+	flushContext := FlushContext{FlushHandle: rspCP.ObjectHandle}
+	defer flushContext.Execute(thetpm)
+
+	contextSave := ContextSave{
+		SaveHandle: rspCP.ObjectHandle,
+	}
+
+	rspCS, err := contextSave.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("ContextSave failed: %v", err)
+	}
+
+	contextLoad := ContextLoad{
+		Context: rspCS.Context,
+	}
+
+	rspCL, err := contextLoad.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("ContextLoad failed: %v", err)
+	}
+
+	if !cmp.Equal(rspCL.LoadedHandle.KnownName(), rspCP.ObjectHandle.KnownName()) {
```

Comment:
KnownName is for when the caller knows the expected name of the object; calling it on a loaded handle should produce nil.

Instead, can you call ReadPublic on the `rspCL.LoadedHandle` and compare it to `rspCP.Name`?

Reply:
Alright I will make that change now.